src/main.rs

```diff
@@ -31,6 +42,27 @@ struct UsageError {
     message: String,
 }
 
+/// Parses the value of a flag that takes a duration.
+///
+/// At the moment, the only supported unit for the duration are seconds and the textual value
+/// must include the "ms" suffix to support adding extra units in the future if desired.
+fn parse_duration(value: &str) -> Result<Timespec, UsageError> {
+    if !value.ends_with(SECONDS_SUFFIX) {
+        let message = format!("invalid time specification {}: must end in 's'", value);
```

Comment:
If you are going to put `SECONDS_SUFFIX` in a constant I would also use the constant in the error string so that they stay in sync.

Reply:
Heh, of course. The constant was a later addition and apparently I missed this case. Irrelevant after all other changes though.